### PR TITLE
Add basic support for chaining exceptions.

### PR DESCRIPTION
**Pull request by joncham**
Add embedding API to avoid the standard runtime process exit path. This allows us to get some data on crashes until proper support can be added.

### CODE REVIEW

File: src/coreclr/vm/exceptionhandling.cpp [hunk 1]
```diff
@@ -1045,21 +1046,26 @@ ProcessCLRException(IN     PEXCEPTION_RECORD   pExceptionRecord,
...
                 {
-                    EX_TRY
-                    {
-                        GetExceptionMessage(oThrowable, message);
-                    }
-                    EX_CATCH
-                    {
-                    }
-                    EX_END_CATCH(SwallowAllExceptions);
+                   EX_TRY
```
Comment (joncham) [inline src/coreclr/vm/exceptionhandling.cpp]: White space, I'll fix this.

**Review by LeeHammerton — APPROVED** [review]
Seems ok to me.

**Comment by LeeHammerton** [comment]
Dumb Q, but is there any test for this? perhaps its coming later in the unity repo?

**Review by UnityAlex — APPROVED** [review]

**Comment by UnityAlex** [comment]
> Dumb Q, but is there any test for this? perhaps its coming later in the unity repo?

I had a similar thought. I know we have some tests over on the Unity side that check for callstacks but they might be expecting mixed ones 🤔

**Comment by joncham** [comment]
No tests in this repo. I am looking at if I can enable any on the Unity side.

**Pull request merged**